### ISSUE TITLE
Decouple MQTT and USB code

### ISSUE THREAD

**Issue by gbbirkisson** [karloygard/xcomfortd-go#3]
Hey,

So if we want to create an API for xcomfortd and then create a proper HA integration, we somehow need to be able to send data from the USB stick to either MQTT part or the API part of the code, or maybe both.

I suggest that we create a interface that the USB part sends data to. Then MQTT and API part implement that interface.

An added benefit to this is that testing for MQTT and the API would be easier since it does not require the USB stick plugged in while developing.

**Comment by karloygard** [comment]
FYI: I cloned your addon code to https://github.com/karloygard/hassio-addons and expanded the HA discovery code a little. It seems to work reasonably well and has received some testing by another user as well.

I agree that making a proper HA integration would be preferable, have you thought any further about how to do that? You appear to be more knowledgeable about those things than me.

**Comment by gbbirkisson** [comment]
Yeah, we need to make an API in xComfortd that uses websockets to push events back and fourth. I will try to take some time to describe what I think we need to do soon :)

When that is done we can make an integration in Home Assistant and make a pull request to the HA repo so it a seemless part of Home Assistant in the future.

**Comment by karloygard** [comment]
As far as I can tell, the Eaton xComfort Bridge is a completely different device from those supported by this code.

**Comment by oywino** [comment]
> As far as I can tell, the Eaton xComfort Bridge is a completely different device from those supported by this code.

And I think you are right. My understanding is that Eatons marketing announcements now indicates that their Bridge is the future and that Home Controller is likely to be phased out at some point. How about considering rewriting your code to support the Bridge?  Regretfully I'm not a programmer myself, otherwise I would have done it. 🙄

Do you think it is realistic to make the CKOZ-00/14 coexist with the Bridge using your HA Add-on code as-is, assuming it is possible to generate a correct Datapoint-list file?
